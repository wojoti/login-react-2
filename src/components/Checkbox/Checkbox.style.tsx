import styled from "styled-components";
export const Wrapper = styled.input`
  margin-right: 10px;
  accent-color: #f43f5e;
  width: 1rem;
  height: 1rem;
`;
