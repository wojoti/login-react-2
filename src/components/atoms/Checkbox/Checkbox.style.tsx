import styled from 'styled-components';

const StyledCheckbox = styled.input`
  margin-right: 10px;
  accent-color: #f43f5e;
  width: 1rem;
  height: 1rem;
`;
export default StyledCheckbox;
