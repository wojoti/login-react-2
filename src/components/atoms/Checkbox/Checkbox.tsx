import { forwardRef } from "react";
import StyledCheckbox from "./Checkbox.style";

export interface CheckboxProps {
  name: string;
  id: string;
  testId?: string;
}

export type CheckboxRef = HTMLInputElement;

const Checkbox = forwardRef<CheckboxRef, CheckboxProps>(
  ({ name, id, testId }, ref) => (
    <StyledCheckbox
      type="checkbox"
      name={name}
      id={id}
      data-testid={testId}
      ref={ref}
    />
  )
);
Checkbox.displayName = "Checkbox";

export default Checkbox;
