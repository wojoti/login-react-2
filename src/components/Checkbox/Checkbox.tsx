import React from "react";
import { Wrapper } from "./Checkbox.style";

type Props = {
  name?: string;
  id?: string;
  value?: string;
  onFieldChange?: (e: any) => void;
};

const Checkbox = (props: Props) => {
  return (
    <Wrapper
      type="checkbox"
      name={props.name}
      id={props.id}
      value={props.value}
      onChange={props.onFieldChange}
    ></Wrapper>
  );
};

export default Checkbox;
