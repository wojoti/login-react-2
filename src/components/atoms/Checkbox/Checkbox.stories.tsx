/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Checkbox from "./Checkbox";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Checkbox",
  component: Checkbox,
  decorators: [
    (Story) => (
      <div style={{ display: "flex", justifyContent: "center" }}>
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Checkbox>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Checkbox> = (args) => (
  <Checkbox {...args} />
);

export const RememberMe = Template.bind({});
RememberMe.args = {
  name: "remember",
  id: "remember",
};
