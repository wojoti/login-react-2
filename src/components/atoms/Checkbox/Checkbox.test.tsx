import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Checkbox, { CheckboxProps } from "./Checkbox";

afterEach(() => {
  cleanup();
});

const props: CheckboxProps = {
  name: "test checkbox",
  id: "test-checkbox-id",
  testId: "test-checkbox-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<Checkbox {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render checkbox", () => {
  render(<Checkbox {...props} />);
  const checkboxElement = screen.getByTestId("test-checkbox-id");
  expect(checkboxElement).toBeInTheDocument();
});

test("should render checkbox unchecked by default", () => {
  render(<Checkbox {...props} />);
  const checkboxElement = screen.getByTestId("test-checkbox-id");
  expect(checkboxElement).toBeInTheDocument();
  expect(checkboxElement).not.toBeChecked();
});

test("should switch checkbox status after click", () => {
  render(<Checkbox {...props} />);
  const checkboxElement = screen.getByTestId("test-checkbox-id");
  expect(checkboxElement).toBeInTheDocument();
  fireEvent.click(checkboxElement);
  expect(checkboxElement).toBeChecked();
});
