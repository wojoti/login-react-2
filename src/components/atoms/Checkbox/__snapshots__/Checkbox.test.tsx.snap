// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<input
  className="sc-bcXHqe jnPdKK"
  data-testid="test-checkbox-id"
  id="test-checkbox-id"
  name="test checkbox"
  type="checkbox"
/>
`;
